openmrs-keycloak-smart-auth/src/main/java/org/openmrs/contrib/keycloak/smart/auth/provider/SmartLaunchAuthenticator.java

```diff
@@ -78,6 +98,36 @@ public void authenticate(AuthenticationFlowContext context) {
 			patientSelectionUrl = DEFAULT_PATIENT_SELECTION_APP_URL;
 		}
 
+		String scope = context.getAuthenticationSession().getClientNote(OIDCLoginProtocol.SCOPE_PARAM);
+		Map<String, ClientScopeModel> defaultScopes = context.getAuthenticationSession().getClient()
+				.getClientScopes(true, true);
+
+		if (scope == null) {
+			scope = "";
+		}
+
+		List<String> scopes = Arrays.stream(scope.split("\\s+")).collect(Collectors.toList());
+		String supportedParams = context.getAuthenticatorConfig().getConfig()
+				.get(SmartLaunchAuthenticatorFactory.CONFIG_EXTERNAL_SMART_LAUNCH_SUPPORTED_PARAMS);
+		if (supportedParams == null || supportedParams.isEmpty()) {
+			context.attempted();
+			return;
+		}
+
+		List<String> params = Arrays.stream(supportedParams.split("\\s+")).collect(Collectors.toList());
+		boolean foundMatch = false;
+		for (String param : params) {
+			if (scopes.contains(LAUNCH_SCOPE_PREFIX + param) ||
+					defaultScopes.containsKey(LAUNCH_SCOPE_PREFIX + param)) {
+				foundMatch = true;
+			}
+		}
+
+		if (!foundMatch) {
+			context.attempted();
```

Comment:
```suggestion
			context.success();
```

This will pass the user through even if we don't have a launch scope, etc.